### PR TITLE
VP-4688 Extend Contact model with Title field

### PR DESCRIPTION
**Pull request by trueboroda**

### CODE REVIEW

**Review by OlegoO — DISMISSED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/module.manifest [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <platformVersion>3.0.0</platformVersion>
     <dependencies>
 <!--        <dependency id="VirtoCommerce.Core" version="3.0.0" /> -->
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/module.manifest]: Remove commented code

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoRepository.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using System.Threading.Tasks;
+using VirtoCommerce.CustomerModule.Data.Model;
+using VirtoCommerce.CustomerModule.Data.Repositories;
+using VirtoCommerce.DemoSolutionFeaturesModule.Data.Models;
+
+namespace VirtoCommerce.DemoSolutionFeaturesModule.Data.Repositories
+{
+    public class CustomerDemoRepository : CustomerRepository
+    {
+        public CustomerDemoRepository(CustomerDemoDbContext dbContext) : base(dbContext)
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoRepository.cs]: Call base constructor on new line

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoRepository.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoRepository.cs:1]: Remove unused namespaces

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoDbContext.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+namespace VirtoCommerce.DemoSolutionFeaturesModule.Data.Repositories
+{
+    public class CustomerDemoDbContext : CustomerDbContext
+    {
+        public CustomerDemoDbContext(DbContextOptions<CustomerDemoDbContext> options)
+          : base(options)
+        {
+        }
+
+        protected CustomerDemoDbContext(DbContextOptions options)
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoDbContext.cs]: Are we really need overloaded protected constructor?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        public override MemberEntity FromModel(Member member, PrimaryKeyResolvingMap pkMap)
+        {
+            if (member is ContactDemo contact)
+            {
+                Title = contact.Title;
+            }
+
+            //Call base converter
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs]: Why this comment here?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public class ContactDemoEntity : ContactEntity
+    {
+        [StringLength(128)]
+        public string Title { get; set; }
+
+
+        public override Member ToModel(Member member)
+        {
+            //Call base converter first
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs]: Remove redundant comment and add empty line after call base method

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+using VirtoCommerce.DemoSolutionFeaturesModule.Core.Models;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.DemoSolutionFeaturesModule.Data.Models
+{
+    public class ContactDemoEntity : ContactEntity
+    {
+        [StringLength(128)]
+        public string Title { get; set; }
+
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs:13]: Remove redundant empty line. Double empty lines are using for separate protected and private methods from public

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Migrations/20200924135525_ChangeContactEntityDiscriminatorToContactDemoEntity.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace VirtoCommerce.DemoSolutionFeaturesModule.Data.Migrations
+{
+    public partial class ChangeContactEntityDiscriminatorToContactDemoEntity : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql(
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Migrations/20200924135525_ChangeContactEntityDiscriminatorToContactDemoEntity.cs]: Use single line SQL and add space after WHERE

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Migrations/20200923061422_InitialContactDemo.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace VirtoCommerce.DemoSolutionFeaturesModule.Data.Migrations
+{
+    public partial class InitialContactDemo : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Migrations/20200923061422_InitialContactDemo.cs]: Why we need empty migration?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Core/Models/ContactDemo.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using VirtoCommerce.CustomerModule.Core.Model;
+
+namespace VirtoCommerce.DemoSolutionFeaturesModule.Core.Models
+{
+    public class ContactDemo: Contact
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Core/Models/ContactDemo.cs]: Add space before ":"

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            if (member is ContactDemo contact)
+            {
+                Title = contact.Title;
+            }
+
+            //Call base converter
+            return base.FromModel(member, pkMap);
+        }
+
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Models/ContactDemoEntity.cs:36]: One empty line

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoDbContext.cs [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+            : base(options)
+        {
+        }
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoDbContext.cs:14]: Add empty line

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoDbContext.cs (hunk 4 above)
Comment (trueboroda) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Repositories/CustomerDemoDbContext.cs]: No. It was already here.

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Migrations/20200923061422_InitialContactDemo.cs (hunk 9 above)
Comment (trueboroda) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Migrations/20200923061422_InitialContactDemo.cs]: It was removed.

**Review by pushnitsa — DISMISSED** [review]

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**